service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/GeneratedOrderControllerTest.java

```diff
@@ -216,4 +204,104 @@ private CallbackRequest buildCallbackRequest() {
             .put("localAuthorityOrCafcass", LOCAL_AUTHORITY_NAME)
             .build();
     }
+
+    @Nested
+    class MidEvent {
+        private final byte[] pdf = {1, 2, 3, 4, 5};
+        private Document document;
+
+        @BeforeEach
+        void setUp() {
+            document = document();
+            DocmosisDocument docmosisDocument = new DocmosisDocument("order.pdf", pdf);
+
+            given(docmosisDocumentGeneratorService.generateDocmosisDocument(any(), any())).willReturn(docmosisDocument);
+            given(uploadDocumentService.uploadPDF(any(), any(), any(), any())).willReturn(document);
+        }
+
+        @Test
+        void midEventShouldGenerateOrderDocumentWhenOrderTypeIsBlankOrder() {
+            AboutToStartOrSubmitCallbackResponse callbackResponse = postMidEvent(generateBlankOrderCaseDetails());
+
+            CaseData caseData = mapper.convertValue(callbackResponse.getData(), CaseData.class);
+
+            verify(uploadDocumentService, times(1)).uploadPDF(userId, userAuthToken, pdf, "blank_order_c21.pdf");
+
+            assertThat(caseData.getOrderTypeAndDocument().getDocument()).isEqualTo(expectedDocument());
+        }
+
+        @Test
+        void midEventShouldGenerateOrderDocumentWhenOrderTypeIsCareOrderWithFurtherDirections() {
+            final AboutToStartOrSubmitCallbackResponse callbackResponse = postMidEvent(
+                generateCareOrderCaseDetails(true));
```

Comment:
generateCareOrderCaseDetails(true) makes me wanna be sick. What does passing true and false do here? To me it seems like true = I wanna generateCareOrderCaseDetails. If that's the case, I would just call the method. I am sure it works differently to this... but it should be clear so I can use the method without necessarily needing to read how it works

Reply:
Updated to be less nauseous